Tests/HAConnectionImpl.test.swift

```diff
@@ -1461,6 +1461,81 @@ internal class HAConnectionImplTests: XCTestCase {
         let container = connection.caches
         XCTAssertEqual(ObjectIdentifier(container.connection), ObjectIdentifier(connection))
     }
+
+    func testWriteDataRequestAddsToRequestController() {
+        connection.connectAutomatically = true
+        let expectedData = "Fake data".data(using: .utf8)!
+        let request = HARequest(
+            type: .sttData(.init(sttBinaryHandlerId: 1)),
+            data: [
+                "audioData": expectedData.base64EncodedString(),
+            ]
+        )
+        connection.connect()
+        connection.send(request) { _ in }
```

Comment:
I mean this literal closure here. The completion block for send() - can you test it is invoked?

Reply:
The original implementation of the "send" method didn't have any usage of the completion. Should this be altered?